### PR TITLE
Hiding deathmessage in vanish.

### PR DESCRIPTION
**Pull request by Destitution-SC**
Hides your death message if you're in vanish to prevent exposing yourself.

### CODE REVIEW

File: Essentials/src/com/earth2me/essentials/EssentialsPlayerListener.java [hunk 1]
```diff
@@ -95,6 +94,14 @@ public void registerEvents() {
             ess.getServer().getPluginManager().registerEvents(new CommandSendListener(), ess);
         }
     }
+    
+    @EventHandler(priority = EventPriority.NORMAL)
+    public void onPlayerDeath(final PlayerDeathEvent event) {
```
Comment (mdcfe) [inline Essentials/src/com/earth2me/essentials/EssentialsPlayerListener.java:100]: You need to import the event

**Review by mdcfe — CHANGES_REQUESTED** [review]
This PR currently doesn't compile.

**Review by YanisBft — DISMISSED** [review]

File: Essentials/src/com/earth2me/essentials/EssentialsPlayerListener.java [hunk 2]
```diff
@@ -95,6 +95,14 @@ public void registerEvents() {
...
+        final User user = ess.getUser((Player) event.getEntity());
+        if(user.isVanished()) {
```
Comment (mdcfe) [inline Essentials/src/com/earth2me/essentials/EssentialsPlayerListener.java]:
```suggestion
        if (user.isVanished()) {
```

**Review by mdcfe — CHANGES_REQUESTED** [review]

**Review by mdcfe — CHANGES_REQUESTED** [review]
IMO this should be behind a config option. Some people might want to give people vanish without keeping vanished players a complete secret.

File: Essentials/src/com/earth2me/essentials/EssentialsPlayerListener.java [hunk 3]
```diff
@@ -95,6 +95,14 @@ public void registerEvents() {
             ess.getServer().getPluginManager().registerEvents(new CommandSendListener(), ess);
         }
     }
+    
+    @EventHandler(priority = EventPriority.NORMAL)
+    public void onPlayerDeath(final PlayerDeathEvent event) {
+        final User user = ess.getUser((Player) event.getEntity());
```
Comment (JRoy) [inline Essentials/src/com/earth2me/essentials/EssentialsPlayerListener.java:101]: Why is there a cast here, it's already a player.

**Review by JRoy — COMMENTED** [review]
Agree with @md678685, needs a config option

**Comment by pop4959** [comment]
I would suggest instead adding a permission that shows the death message (when negated, it hides it). This creates a more robust and generalized solution, where the death messages can be conditionally hidden, and for vanish, you would only need something like the following:
`/lp group default permission set essentials.showdeathmessage false essentials:vanished=true`

**Comment by pop4959** [comment]
Closing due to author inactivity. Can re-open if there are signs of life, especially addressing any concerns that have been mentioned.

**Pull request closed without merging**